src/swsssdk/configdb.py

```diff
@@ -328,10 +328,135 @@ def get_config(self):
                 key = key.decode('utf-8')
             try:
                 (table_name, row) = key.split(self.TABLE_NAME_SEPARATOR, 1)
-                entry = self.__raw_to_typed(client.hgetall(key))
+                entry = self.raw_to_typed(client.hgetall(key))
                 if entry != None:
                     data.setdefault(table_name, {})[self.deserialize_key(row)] = entry
             except ValueError:
                 pass    #Ignore non table-formated redis entries
         return data
 
+
+class ConfigDBPipeConnector(ConfigDBConnector):
```

Comment:
>ConfigDBPipeConnector [](start = 6, length = 21)

From the design point of view, ConfigDBPipeConnector is not a subclass of ConfigDBConnector (ConfigDBNonPipeConnector).

I suggest reuse original class.
```
def __init__(self, use_pipeline=False, **kwargs):
```

Reply:
it is extending ConfigDBConnector functionality by providing pipeline based communication for mod_config, get_config. the use_pipeline switch resemble the use_unix_socket_path option. However, the use_pipeline is propagating to the connector class. This will result in if statement or maintaining internal vptr table which is done by subclassing in this case.

Reply:
> However, the use_pipeline is propagating to the connector class

This is not true. you can check use_pipeline first and remove it from kwargs.

---
In reply to: [459153837](https://github.com/Azure/sonic-py-swsssdk/pull/83#discussion_r459153837) [](ancestors = 459153837)

Reply:
Correct. What I meant is that use_pipeline switch is not the same as use_unix_socket _domain. It is use in this class will require if statement or creating vptr table which is given for free when extend the class.